src/template/span.rs

```diff
@@ -1,71 +1,75 @@
 use std::fmt;
-use std::ops::{Deref, DerefMut, Index};
-
-type BytePosType = u32;
-
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct BytePos(pub BytePosType);
-
-impl BytePos {
-	pub fn new(value: BytePosType) -> Self {
-		Self(value)
-	}
-
-	pub fn from_usize(value: usize) -> Self {
-		Self(value as BytePosType)
-	}
-
-	pub fn as_usize(&self) -> usize {
-		self.0 as usize
-	}
-
-	pub fn into_inner(self) -> BytePosType {
-		self.0
-	}
-}
-
-impl fmt::Display for BytePos {
-	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-		fmt::Display::fmt(&self.0, f)
-	}
-}
-
-impl From<usize> for BytePos {
-	fn from(value: usize) -> Self {
-		BytePos::from_usize(value)
-	}
-}
-
-impl From<BytePosType> for BytePos {
-	fn from(value: BytePosType) -> Self {
-		Self(value)
-	}
-}
-
-impl From<BytePos> for usize {
-	fn from(value: BytePos) -> Self {
-		value.as_usize()
-	}
-}
-
-impl From<BytePos> for BytePosType {
-	fn from(value: BytePos) -> Self {
-		value.0
-	}
-}
-
-impl Deref for BytePos {
-	type Target = BytePosType;
-
-	fn deref(&self) -> &Self::Target {
-		&self.0
-	}
-}
-
-impl DerefMut for BytePos {
-	fn deref_mut(&mut self) -> &mut Self::Target {
-		&mut self.0
-	}
+use std::ops::{Deref, Index};
+
+pub trait Pos {
```

Comment:
Position? or what should it mean? comments would be great

Reply:
I will add some basic doc comments. I plan to extensively comment all code when #25 is merged.

Reply:
noice